### ISSUE TITLE
The TypeScript language service died 5 times right after it got started. The service will not be restarted.

### ISSUE THREAD

**Issue by David-Else** [axetroy/vscode-deno#86]
**Describe the bug**

When untitled files are created with ctrl-n and deno extension is running then error `The TypeScript language service died 5 times right after it got started. The service will not be restarted.` This does not happen on all projects and seems kind of random.

This was a bug in TypeScript before 3.8.2, could it be linked?

https://github.com/microsoft/vscode/issues/91027#issuecomment-592681683

I had it a lot, but it was definately fixed 3.8.2

here is typescript logs, I can see errors happening, do they make sense?
```
[typescript-deno-plugin] "compilationSettings:{\"allowJs\":true,\"checkJs\":false,\"strict\":true,\"esModuleInterop\":true,\"jsx\":2,\"module\":99,\"moduleResolution\":2,\"outDir\":\"$deno$\",\"resolveJsonModule\":true,\"sourceMap\":true,\"stripComments\":true,\"target\":3,\"noEmit\":true,\"noEmitHelpers\":true,\"allowSyntheticDefaultImports\":true,\"allowNonTsExtensions\":true,\"noEmitForJsFiles\":true}"
Err 32    [20:16:16.558] Exception on executing command {"seq":2,"type":"request","command":"updateOpen","arguments":{"changedFiles":[],"closedFiles":[],"openFiles":[{"file":"untitled:^Untitled-1","fileContent":"let x = 11;\n\nconsole.log(x)\n\n","projectRootPath":"/home/david/sites/TypeScript/eslint-config-octopus-GITHUB","scriptKindName":"TS"},{"file":"untitled:^Untitled-2","fileContent":"let xd d d f","projectRootPath":"/home/david/sites/TypeScript/eslint-config-octopus-GITHUB","scriptKindName":"TS"}]}}:

    The expression evaluated to a falsy value:
    
      assert_1.default(path.isAbsolute(containingFile))
    

    AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:
```
full log...
```
Info 0    [20:16:16.474] Starting TS Server
Info 1    [20:16:16.474] Version: 3.8.3
Info 2    [20:16:16.474] Arguments: /usr/share/code/code /home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/typescript/lib/tsserver.js --useInferredProjectPerProjectRoot --enableTelemetry --cancellationPipeName /tmp/vscode-typescript1000/8c1b8708ce915b9f58bd/tscancellation-579e26ce505993ed69ed.tmp* --logVerbosity verbose --logFile /home/david/.config/Code/logs/20200302T201613/exthost1/vscode.typescript-language-features/tsserver-log-sB9N1d/tsserver.log --globalPlugins typescript-vscode-sh-plugin,typescript-deno-plugin --pluginProbeLocations /usr/share/code/resources/app/extensions/typescript-language-features,/home/david/.vscode/extensions/axetroy.vscode-deno-3.0.5 --locale en --noGetErrOnBackgroundUpdate --validateDefaultNpmLocation
Info 3    [20:16:16.474] Platform: linux NodeVersion: 12 CaseSensitive: true
Info 4    [20:16:16.476] Binding...
Info 5    [20:16:16.481] request:
    {"seq":0,"type":"request","command":"configure","arguments":{"hostInfo":"vscode","preferences":{"providePrefixAndSuffixTextForRename":true,"allowRenameOfImportPath":true}}}
Info 6    [20:16:16.481] Host information vscode
Info 7    [20:16:16.481] response:
    {"seq":0,"type":"response","command":"configure","request_seq":0,"success":true}
Perf 8    [20:16:16.482] 0::configure: async elapsed time (in milliseconds) 1.2212
Info 9    [20:16:16.482] request:
    {"seq":1,"type":"request","command":"compilerOptionsForInferredProjects","arguments":{"options":{"module":"commonjs","target":"es2016","jsx":"preserve","sourceMap":true,"allowJs":true,"allowSyntheticDefaultImports":true,"allowNonTsExtensions":true}}}
Info 10   [20:16:16.483] Scheduled: *ensureProjectForOpenFiles*
Perf 11   [20:16:16.483] 1::compilerOptionsForInferredProjects: elapsed time (in milliseconds) 0.7713
Info 12   [20:16:16.483] response:
    {"seq":0,"type":"response","command":"compilerOptionsForInferredProjects","request_seq":1,"success":true,"body":true}
Info 13   [20:16:16.483] request:
    {"seq":2,"type":"request","command":"updateOpen","arguments":{"changedFiles":[],"closedFiles":[],"openFiles":[{"file":"untitled:^Untitled-1","fileContent":"let x = 11;\n\nconsole.log(x)\n\n","projectRootPath":"/home/david/sites/TypeScript/eslint-config-octopus-GITHUB","scriptKindName":"TS"},{"file":"untitled:^Untitled-2","fileContent":"let xd d d f","projectRootPath":"/home/david/sites/TypeScript/eslint-config-octopus-GITHUB","scriptKindName":"TS"}]}}
Info 14   [20:16:16.484] Search path: 
Info 15   [20:16:16.484] For info: untitled:^Untitled-1 :: No config files found.
Info 16   [20:16:16.486] Loading global plugin typescript-vscode-sh-plugin
Info 17   [20:16:16.486] Enabling plugin typescript-vscode-sh-plugin from candidate paths: /usr/share/code/resources/app/extensions/typescript-language-features,/home/david/.vscode/extensions/axetroy.vscode-deno-3.0.5,/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js/../../..
Info 18   [20:16:16.486] Loading typescript-vscode-sh-plugin from /usr/share/code/resources/app/extensions/typescript-language-features (resolved to /usr/share/code/resources/app/extensions/typescript-language-features/node_modules)
Info 19   [20:16:16.493] typescript-vscode-sh-plugin initialized. Intercepting getEncodedSemanticClassifications and getEncodedSyntacticClassifications.
Info 20   [20:16:16.493] Plugin validation succeded
Info 21   [20:16:16.493] Loading global plugin typescript-deno-plugin
Info 22   [20:16:16.493] Enabling plugin typescript-deno-plugin from candidate paths: /usr/share/code/resources/app/extensions/typescript-language-features,/home/david/.vscode/extensions/axetroy.vscode-deno-3.0.5,/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js/../../..
Info 23   [20:16:16.494] Loading typescript-deno-plugin from /usr/share/code/resources/app/extensions/typescript-language-features (resolved to /usr/share/code/resources/app/extensions/typescript-language-features/node_modules)
Info 24   [20:16:16.510] Loading typescript-deno-plugin from /home/david/.vscode/extensions/axetroy.vscode-deno-3.0.5 (resolved to /home/david/.vscode/extensions/axetroy.vscode-deno-3.0.5/node_modules)
Info 25   [20:16:16.537] [typescript-deno-plugin] "Create typescript-deno-plugin"
Info 26   [20:16:16.537] Plugin validation succeded
Info 27   [20:16:16.538] [typescript-deno-plugin] "compilationSettings:{\"allowJs\":true,\"checkJs\":false,\"strict\":true,\"esModuleInterop\":true,\"jsx\":2,\"module\":99,\"moduleResolution\":2,\"outDir\":\"$deno$\",\"resolveJsonModule\":true,\"sourceMap\":true,\"stripComments\":true,\"target\":3,\"noEmit\":true,\"noEmitHelpers\":true,\"allowSyntheticDefaultImports\":true,\"allowNonTsExtensions\":true,\"noEmitForJsFiles\":true}"
Info 28   [20:16:16.538] Starting updateGraphWorker: Project: /dev/null/inferredProject1*
Info 29   [20:16:16.540] FileWatcher:: Added:: WatchInfo: /home/david/.cache/deno/lib.deno_runtime.d.ts 500 undefined Project:  WatchType: Closed Script info
Info 30   [20:16:16.540] [typescript-deno-plugin] "compilationSettings:{\"allowJs\":true,\"checkJs\":false,\"strict\":true,\"esModuleInterop\":true,\"jsx\":2,\"module\":99,\"moduleResolution\":2,\"outDir\":\"$deno$\",\"resolveJsonModule\":true,\"sourceMap\":true,\"stripComments\":true,\"target\":3,\"noEmit\":true,\"noEmitHelpers\":true,\"allowSyntheticDefaultImports\":true,\"allowNonTsExtensions\":true,\"noEmitForJsFiles\":true}"
Info 31   [20:16:16.541] [typescript-deno-plugin] "compilationSettings:{\"allowJs\":true,\"checkJs\":false,\"strict\":true,\"esModuleInterop\":true,\"jsx\":2,\"module\":99,\"moduleResolution\":2,\"outDir\":\"$deno$\",\"resolveJsonModule\":true,\"sourceMap\":true,\"stripComments\":true,\"target\":3,\"noEmit\":true,\"noEmitHelpers\":true,\"allowSyntheticDefaultImports\":true,\"allowNonTsExtensions\":true,\"noEmitForJsFiles\":true}"
Err 32    [20:16:16.558] Exception on executing command {"seq":2,"type":"request","command":"updateOpen","arguments":{"changedFiles":[],"closedFiles":[],"openFiles":[{"file":"untitled:^Untitled-1","fileContent":"let x = 11;\n\nconsole.log(x)\n\n","projectRootPath":"/home/david/sites/TypeScript/eslint-config-octopus-GITHUB","scriptKindName":"TS"},{"file":"untitled:^Untitled-2","fileContent":"let xd d d f","projectRootPath":"/home/david/sites/TypeScript/eslint-config-octopus-GITHUB","scriptKindName":"TS"}]}}:

    The expression evaluated to a falsy value:
    
      assert_1.default(path.isAbsolute(containingFile))
    

    AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:
    
      assert_1.default(path.isAbsolute(containingFile))
    
        at new ModuleResolver (/home/david/.vscode/extensions/axetroy.vscode-deno-3.0.5/node_modules/typescript-deno-plugin/out/core/module_resolver.js:28:25)
        at Function.create (/home/david/.vscode/extensions/axetroy.vscode-deno-3.0.5/node_modules/typescript-deno-plugin/out/core/module_resolver.js:31:16)
        at InferredProject.info.languageServiceHost.resolveTypeReferenceDirectives (/home/david/.vscode/extensions/axetroy.vscode-deno-3.0.5/node_modules/typescript-deno-plugin/out/typescript-deno-plugin/src/plugin.js:120:67)
        at Object.compilerHost.resolveTypeReferenceDirectives (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:135201:64)
        at resolveTypeReferenceDirectiveNamesWorker (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:97730:137)
        at processTypeReferenceDirectives (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:99347:31)
        at findSourceFile (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:99185:21)
        at /home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:99020:85
        at getSourceFileFromReferenceWorker (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:99005:77)
        at processSourceFile (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:99020:13)
        at processRootFile (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:98843:13)
        at /home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:97809:60
        at Object.forEach (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:167:30)
        at Object.createProgram (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:97809:16)
        at synchronizeHostData (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:135218:26)
        at Proxy.getProgram (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:135310:13)
        at InferredProject.Project.updateGraphWorker (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:139786:53)
        at InferredProject.Project.updateGraph (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:139729:42)
        at ProjectService.assignOrphanScriptInfoToInferredProject (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:141705:25)
        at ProjectService.assignProjectToOpenedScriptInfo (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:143132:26)
        at /home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:143396:107
        at Object.mapDefined (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:562:30)
        at ProjectService.applyChangesInOpenFiles (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:143396:48)
        at Session.handlers.ts.createMapFromTemplate._a.<computed> (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:144260:46)
        at /home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:146003:88
        at IOSession.Session.executeWithRequestId (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:145994:28)
        at IOSession.Session.executeCommand (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:146003:33)
        at IOSession.Session.onMessage (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:146027:35)
        at Interface.<anonymous> (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:147342:27)
        at Interface.emit (events.js:200:13)
        at Interface._onLine (readline.js:314:10)
        at Interface._normalWrite (readline.js:459:12)
        at Socket.ondata (readline.js:170:10)
        at Socket.emit (events.js:200:13)
        at addChunk (_stream_readable.js:294:12)
        at readableAddChunk (_stream_readable.js:275:11)
        at Socket.Readable.push (_stream_readable.js:210:10)
        at Pipe.onStreamRead (internal/stream_base_commons.js:166:17)
Info 33   [20:16:16.558] response:
    {"seq":0,"type":"response","command":"updateOpen","request_seq":2,"success":false,"message":"Error processing request. The expression evaluated to a falsy value:\n\n  assert_1.default(path.isAbsolute(containingFile))\n\nAssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:\n\n  assert_1.default(path.isAbsolute(containingFile))\n\n    at new ModuleResolver (/home/david/.vscode/extensions/axetroy.vscode-deno-3.0.5/node_modules/typescript-deno-plugin/out/core/module_resolver.js:28:25)\n    at Function.create (/home/david/.vscode/extensions/axetroy.vscode-deno-3.0.5/node_modules/typescript-deno-plugin/out/core/module_resolver.js:31:16)\n    at InferredProject.info.languageServiceHost.resolveTypeReferenceDirectives (/home/david/.vscode/extensions/axetroy.vscode-deno-3.0.5/node_modules/typescript-deno-plugin/out/typescript-deno-plugin/src/plugin.js:120:67)\n    at Object.compilerHost.resolveTypeReferenceDirectives (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:135201:64)\n    at resolveTypeReferenceDirectiveNamesWorker (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:97730:137)\n    at processTypeReferenceDirectives (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:99347:31)\n    at findSourceFile (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:99185:21)\n    at /home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:99020:85\n    at getSourceFileFromReferenceWorker (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:99005:77)\n    at processSourceFile (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:99020:13)\n    at processRootFile (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:98843:13)\n    at /home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:97809:60\n    at Object.forEach (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:167:30)\n    at Object.createProgram (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:97809:16)\n    at synchronizeHostData (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:135218:26)\n    at Proxy.getProgram (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:135310:13)\n    at InferredProject.Project.updateGraphWorker (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:139786:53)\n    at InferredProject.Project.updateGraph (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:139729:42)\n    at ProjectService.assignOrphanScriptInfoToInferredProject (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:141705:25)\n    at ProjectService.assignProjectToOpenedScriptInfo (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:143132:26)\n    at /home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:143396:107\n    at Object.mapDefined (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:562:30)\n    at ProjectService.applyChangesInOpenFiles (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:143396:48)\n    at Session.handlers.ts.createMapFromTemplate._a.<computed> (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:144260:46)\n    at /home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:146003:88\n    at IOSession.Session.executeWithRequestId (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:145994:28)\n    at IOSession.Session.executeCommand (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:146003:33)\n    at IOSession.Session.onMessage (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:146027:35)\n    at Interface.<anonymous> (/home/david/sites/TypeScript/eslint-config-octopus-GITHUB/node_modules/.pnpm/registry.npmjs.org/typescript/3.8.3/node_modules/typescript/lib/tsserver.js:147342:27)\n    at Interface.emit (events.js:200:13)\n    at Interface._onLine (readline.js:314:10)\n    at Interface._normalWrite (readline.js:459:12)\n    at Socket.ondata (readline.js:170:10)\n    at Socket.emit (events.js:200:13)\n    at addChunk (_stream_readable.js:294:12)\n    at readableAddChunk (_stream_readable.js:275:11)\n    at Socket.Readable.push (_stream_readable.js:210:10)\n    at Pipe.onStreamRead (internal/stream_base_commons.js:166:17)"}
Info 34   [20:16:16.558] event:
    {"seq":0,"type":"event","event":"typingsInstallerPid","body":{"pid":22538}}
Info 35   [20:16:16.559] request:
    {"seq":3,"type":"request","command":"configurePlugin","arguments":{"pluginName":"typescript-deno-plugin","configuration":{"enable":true,"dts_file":[],"import_map":"import_map.json"}}}
Info 36   [20:16:16.559] [typescript-deno-plugin] "onConfigurationChanged: {\"enable\":true,\"dts_file\":[],\"import_map\":\"import_map.json\"}"
```
```

**Comment by axetroy** [comment]
As we can see from the logs, an exception is thrown for calling `compilerHost.resolveTypeReferenceDirectives`

Here is the logic

https://github.com/axetroy/vscode-deno/blob/c7df99b18e6846218c529f19219fb2bae6f7994a/typescript-deno-plugin/src/plugin.ts#L177-L180

https://github.com/axetroy/vscode-deno/blob/c7df99b18e6846218c529f19219fb2bae6f7994a/core/module_resolver.ts#L34-L40

This is because `containingFile` is not an absolute path

I'm not sure if this is a typescript issue

I will continue to identify the problem

**Comment by axetroy** [comment]
I think one of the possible reason is that you created a new untitled TextDocument

```
AssertionError [ERR_ASSERTION]: ModuleResolver filepath require absolute but got untitled:^Untitled-1
    at new ModuleResolver (/Users/axetroy/gpm/github.com/axetroy/vscode-deno/node_modules/typescript-deno-plugin/out/core/module_resolver.js:28:25)
    at Function.create (/Users/axetroy/gpm/github.com/axetroy/vscode-deno/node_modules/typescript-deno-plugin/out/core/module_resolver.js:31:16)
    at InferredProject.info.languageServiceHost.resolveTypeReferenceDirectives (/Users/axetroy/gpm/github.com/axetroy/vscode-deno/node_modules/typescript-deno-plugin/out/typescript-deno-plugin/src/plugin.js:120:67)
    at Object.compilerHost.resolveTypeReferenceDirectives (tsserver.js:135201:64)
    at resolveTypeReferenceDirectiveNamesWorker (tsserver.js:97730:137)
    at processTypeReferenceDirectives (tsserver.js:99347:31)
    at findSourceFile (tsserver.js:99185:21)
    at tsserver.js:99020:85
    at getSourceFileFromReferenceWorker (tsserver.js:99005:77)
    at processSourceFile (tsserver.js:99020:13)
    at processRootFile (tsserver.js:98843:13)
    at tsserver.js:97809:60
    at Object.forEach (tsserver.js:167:30)
    at Object.createProgram (tsserver.js:97809:16)
    at synchronizeHostData (tsserver.js:135218:26)
    at Proxy.getProgram (tsserver.js:135310:13)
    at InferredProject.Project.updateGraphWorker (tsserver.js:139786:53)
    at InferredProject.Project.updateGraph (tsserver.js:139729:42)
    at ProjectService.assignOrphanScriptInfoToInferredProject (tsserver.js:141705:25)
    at ProjectService.assignProjectToOpenedScriptInfo (tsserver.js:143132:26)
    at tsserver.js:143396:107
    at Object.mapDefined (tsserver.js:562:30)
    at ProjectService.applyChangesInOpenFiles (tsserver.js:143396:48)
    at Session.handlers.ts.createMapFromTemplate._a.<computed> (tsserver.js:144260:46)
    at tsserver.js:146003:88
    at IOSession.Session.executeWithRequestId (tsserver.js:145994:28)
    at IOSession.Session.executeCommand (tsserver.js:146003:33)
    at IOSession.Session.onMessage (tsserver.js:146027:35)
    at Interface.<anonymous> (tsserver.js:147342:27)
    at Interface.emit (events.js:203:13)
    at Interface._onLine (readline.js:316:10)
    at Interface._normalWrite (readline.js:461:12)
    at Socket.ondata (readline.js:172:10)
    at Socket.emit (events.js:203:13)
    at addChunk (_stream_readable.js:295:12)
    at readableAddChunk (_stream_readable.js:276:11)
    at Socket.Readable.push (_stream_readable.js:210:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:166:17)
```

I will try to fix this in next version

**Comment by axetroy** [comment]
Since I don't know if there are other reasons for the crash, I will leave the issue temporarily

**Comment by David-Else** [comment]
It seems fixed in 3.0.6 :) Thanks!

I don't know how your extensions somehow made a bug come back that was fixed, but here is the history of me reporting the bug and the links to the exact fixes in TypeScript that made it go away. All the info you might want should be there:

orginal thread:`Debug failure in ProjectService.updateNonInferredProjectFiles #36200`
https://github.com/microsoft/TypeScript/issues/36200

original fix from `3.8.1`: `Handle untitled files from vscode which are of format: untitled:^Untitled-1`
https://github.com/microsoft/TypeScript/pull/36240

**Comment by axetroy** [comment]
I have found the problem and fixed it in v3.0.6

https://github.com/axetroy/vscode-deno/blob/fd71d0c19ee6b3c957019cb0299e01ffca11051c/typescript-deno-plugin/src/plugin.ts#L169-L173
